src/components/calendar/Month.tsx

```diff
@@ -39,37 +39,50 @@ const calculateWorkingHours = (firstDay: number, lastDay: number) => {
   return val
 }
 
-const Month = ({ currentDate }: any) => {
+const Month = (props: any) => {
   const { toggleBarOnDateClick, isYearView } = useCalendarContext()
 
   // 42 -> 49 to include working hours at the bottom of calendar
   // small chance this number will increase by 7 (6 possible weeks + 1 for the last empty week) to include working week on the side???
-  const numBoxes = 49
+  let numBoxes = 49
 
   // for working hours in a month
   const firstDay = new Date(
-    currentDate.getFullYear(),
-    currentDate.getMonth(),
+    props.currentDate.getFullYear(),
+    props.currentDate.getMonth(),
     1
   ).getDay()
 
   const firstDayOffset = firstDay - 1
 
   const numDaysInMonth = new Date(
-    currentDate.getFullYear(),
-    currentDate.getMonth() + 1,
+    props.currentDate.getFullYear(),
+    props.currentDate.getMonth() + 1,
     0
   ).getDate()
 
   // also for working hours in a month
   const lastDay = new Date(
-    currentDate.getFullYear(),
-    currentDate.getMonth(),
+    props.currentDate.getFullYear(),
+    props.currentDate.getMonth(),
     numDaysInMonth
   ).getDay()
 
   const handleDayClick = (day: any) => {
-    toggleBarOnDateClick(1, formatDate(currentDate, day))
+    toggleBarOnDateClick(1, formatDate(props.currentDate, day))
+  }
+
+  // i do not know why when we switch from month view to year view
+  // the props.yearView is undefined
```

Comment:
if someone knows why this is that would be coolio but idk if it matters that much

Reply:
i can have a lookie later but feel free to merge otherwise